### PR TITLE
Use "e" instead of "b" for opening files, also check that e is a valid file mode

### PR DESCRIPTION
**Pull request by Luciam91**
I think this is all that's required for #19?
Happy to add more if there's a requirement :)

### CODE REVIEW

File: lib/Internal/FileTask.php [hunk 1]
```diff
@@ -65,13 +65,14 @@ public function run(Environment $environment) {
                     case "x+":
                     case "c":
                     case "c+":
+                    case "e":
```
Comment (kelunik) [inline lib/Internal/FileTask.php]: Why did you add `e` here?

File: lib/Internal/FileTask.php [hunk 2]
```diff
@@ -65,13 +65,14 @@ public function run(Environment $environment) {
...
                         break;
 
                     default:
                         throw new \Error("Invalid file mode");
                 }
 
-                $handle = @\fopen($path, $mode . 'b');
+                $handle = @\fopen($path, $mode . 'e');
```
Comment (kelunik) [inline lib/Internal/FileTask.php]: I think it should be `be`.

File: lib/Internal/FileTask.php (hunk 1 above)
Comment (Luciam91) [reply lib/Internal/FileTask.php]: I think I've misinterpreted the code here (and expected that `e` was being passed down, I'll remove :)

**Comment by kelunik** [comment]
@trowski I guess we should also add that to the blocking driver, right? And I think we should have the same `$mode` restrictions there, but that's out of scope for this PR.

**Comment by trowski** [comment]
Manually merged via 271ad2bf4ab97bc01102d8314d511a54180ace59. Also reflected the changes and mode validation into BlockingDriver.
